### ISSUE TITLE
Different spatial resolution leads to different results?

### ISSUE THREAD

**Issue by carme-hp** [maierbn/opendihu#19]
**Setup:**
 I consider a single cuboid muscle with dimensions 3x3x14.8 cm and 4x4 fibers. Each fiber covers the whole length of the muscle(14.8cm) and has 40 points. The fibers are activated in a symmetric fashion using an input file. At z=0, the muscle is fixed to the wall, while at the other end no boundary condition is specified. 

**Goal:**
To investigate what happens when the spatial resolution of the 3D continuum mechanics solver is changed, eg. [mx,my,mz]

**Contraction:**
![contraction](https://user-images.githubusercontent.com/71499004/230307416-4514e778-9c30-436c-9897-7fb345a8451b.png)
I would expect that increasing the resolution would lead to a converged solution (eg. the finer case). I cannot tell if this is what happens here, or whether changing resolution introduces an scaling factor and leads to different results.

**Elongation:**
Now we apply a neumann boundary condition at the z=14.8, such that the muscle is pulled. A linearly increasing traction is applied on the z-component, while the x and y component are set to zero.
The fibers are not activated, so in practice this equivalent to running a mechanics solver alone. 
The plot below shows the results of the elongation. Note that the variance in the muscle deformation is much smaller, even though as in contraction the muscle is changing length for about 0.5cm (I tuned the boundary condition to get a similar deformation in length). 
![elongation](https://user-images.githubusercontent.com/71499004/230311200-fe423ff8-455a-4319-a61c-6a739bee7955.png)

Even though this looks much more like results that converge as the resolution increases, if I zoom in I see the following:
![elongation_zoom](https://user-images.githubusercontent.com/71499004/230311206-b50b66d2-4faa-4c82-8b33-1ba0c24680c9.png)

**Looking for scaling factors:**
I tried investigating if there was a scaling factor. If there was one, I would expect that if I divide the lines above I would get a constant value.

For the contraction case I obtained;
![Figure_3](https://user-images.githubusercontent.com/71499004/230308867-c603f1d8-474c-4747-bd46-7498e8ad1ad8.png)
![Figure_4](https://user-images.githubusercontent.com/71499004/230309226-9bde7ea8-6225-41c1-abfb-4aff1b32b756.png)

For the elongation case I obtained:
![Figure_13](https://user-images.githubusercontent.com/71499004/230314292-e9772205-e4bc-4e25-9fb7-eb66e099be0f.png)
![Figure_14](https://user-images.githubusercontent.com/71499004/230314297-664db30d-7a57-4d88-a73a-b2ab19d17851.png)

**Conclusions / open questions:**
- Different spatial resolution leads to different results
- The difference in results is much higher if fibers are activated.
- But I think that the difference in results should be similar in the contraction and elongation scenario I considered.
- Looking at the plots for scaling in the elongation, all we see are oscillations. I think we can rule out that the mechanics solver introduces any scaling factor. 
- Looking at the plots for scaling in the contraction, we don't see any constant scaling factor yet something is happening. Maybe there is some kind of exponential factor?
- I think something is wrong with the mapping from the fibers to the continuum mechanics mesh, which causes different results. 
- I wonder if the resolution I choose for the fibers makes sense. According to what I have read in BM thesis it should work, but it could be worth to check what happens if I double the fibers resolution...

**Comment by maierbn** [comment]
* I would also rule out the constant factor. I remember I also did some scaling tests of only solid mechanics and compared it to another solver (Febio) and the values looked good. I also can't image that there is an exponential error or anything like that, just trust the implementation.
* Maybe can you try to uniformly refine the mesh and try higher element numbers, e.g. `[mx, mx, 10*mx]` for `mx` from 2 to 10? You can also run it in parallel to save runtime. Then, the convergence should be visible more clearly (hopefully, if any). In the industry, we use hundreds of thousand nodes in a converged setting of structural mechanics FEM (more complex geometries though :-))
* The case with included fibers is clearly more difficult because of the mapping. Maybe you can try a refinement `[4,4,mz]` or similar to remove smoothing effects in x and y direction.
* The fiber resolution of 40 nodes on 14.8 cm seems it bit too coarse for me, I usually used 1480 nodes. But as long as you get a propagating action potential and you're not concerned with the error of the electrophysiology model you can keep this setting.

**Comment by carme-hp** [comment]
I close the issue as the remaining open questions have been better summarized in the new issue #21 .
